src/v2/Apps/Show/Components/ShowNavigationBanner.tsx

```diff
@@ -11,17 +12,28 @@ interface ShowNavigationBannerProps {
 const ShowNavigationBanner: React.FC<ShowNavigationBannerProps & BoxProps> = ({
   show,
 }) => {
+  const { match } = useRouter()
+  const { from_fair } = match.location.query
   const { fair, partner } = show
+  let link = fair?.href
+
+  /**
+   * useMemo is used to ensure that the banner does not disappear immediately
+   * after clicking on the back link, since `from_fair` query param will be removed from url
+   */
+  // eslint-disable-next-line react-hooks/exhaustive-deps
+  const arrivedFromFair = useMemo(() => from_fair ?? false, [])
+
+  if (arrivedFromFair) {
+    link = `${fair?.href}/exhibitors?focused_exhibitor=${partner?.internalID}`
```

Comment:
maybe it would be better to rename `arrivedFromFair` to `arrivedFromExhibitorsTabs` and `from_fair` query param to `from_exhibitors_tab` to make it clearer? 🤔

Reply:
Instead of requiring the ShowApp to construct these "Back to fair" URL overrides, could the full return link be encoded in the show page URL as a query param?

**Fair page exhibitor's tab**

```
back_to_fair_href: /fair/:slug/exhibitors?focused_exhibitor=:partner_id
```

**Fair page overview tab**

```
back_to_fair_href: /fair/:slug?sort=NAME_ASC&page=2
```

The banner in the show app could default to `/fair/:slug` if `isFairBooth=true` and no other information is available. If a `back_to_fair_href` query param, use that value instead for the banner href.

Reply:
@dblandin good idea!

Reply:
Changes to pass `/fair/:slug?sort=NAME_ASC&page=2` to `back_to_fair_href` I will add in a separate PR

Reply:
PR reference: artsy/force#10161